src/NzbDrone.Core/DecisionEngine/Specifications/CutoffSpecification.cs

```diff
@@ -50,23 +50,26 @@ public virtual Decision IsSatisfiedBy(RemoteAlbum subject, SearchCriteriaBase se
 
                 if (!tracksMissing && trackFiles.Any())
                 {
-                    var lowestQuality = trackFiles.Select(c => c.Quality).OrderBy(c => c.Quality.Id).First();
+                    // Get a distinct list of all current track qualities and languages for a given album
+                    var currentQualities = trackFiles.Select(c => c.Quality).Distinct().ToList();
+                    var currentLanguages = trackFiles.Select(c => c.Language).Distinct().ToList();
 
-                    _logger.Debug("Comparing file quality and language with report. Existing file is {0}", lowestQuality.Quality);
+                    _logger.Debug("Comparing file quality and language with report. Existing files contain {0} : {1}", currentQualities, currentLanguages);
```

Comment:
This logging needs fixing - `currentQualities` and `currentLanguages` are lists so don't do `ToString` nicely